### PR TITLE
Add nuget ecosystem

### PR DESCRIPTION
**Pull request by jpopelka**
- [x] CVEcheckerTask
- [x] DependencySnapshotTask
- [x] InitAnalysisFlow
- [x] LicenseCheckTask
- [x] MercatorTask

### CODE REVIEW

**Comment by pkajaba** [comment]
[test this]

**Comment by pkajaba** [comment]
Sorry for closing, it was missclick

**Comment by jpopelka** [comment]
[test this]

**Comment by jpopelka** [comment]
[test this]

File: f8a_worker/process.py [hunk 1]
```diff
@@ -363,6 +364,17 @@ def release_key(rel):
             git = Git.clone(artifact, target_dir)
             digest = IndianaJones.get_revision(target_dir)
             artifact_path = git.archive(artifact)
+        elif ecosystem.is_backed_by(EcosystemBackend.nuget):
+            git = Git.create_git(target_dir)
+            # https://api.nuget.org/packages/windowsazure.storage.8.1.0.nupkg
```
Comment (pkajaba) [inline f8a_worker/process.py]: Is this comment necessary?

**Comment by pkajaba** [comment]
[test this]

**Comment by pkajaba** [comment]
[test this]

File: alembic/versions/a31c98c74ea0_add_e_p_v.py [hunk 2]
```diff
@@ -34,6 +34,7 @@ def upgrade():
             {'id': 4, 'name': 'pypi', '_backend': 'pypi', 'url': 'https://pypi.python.org/', 'fetch_url': 'https://pypi.python.org/pypi'},
             {'id': 5, 'name': 'go', '_backend': 'scm', 'url': None, 'fetch_url': None},
             {'id': 6, 'name': 'crates', '_backend': 'crates', 'url': 'https://crates.io/', 'fetch_url': None},
+            {'id': 7, 'name': 'nuget', '_backend': 'nuget', 'url': 'https://nuget.org/', 'fetch_url': 'https://api.nuget.org/packages/'},
```
Comment (pkajaba) [inline alembic/versions/a31c98c74ea0_add_e_p_v.py]: Is this really nuget URL? If I try to access it, it returns InvalidUri error

Comment (jpopelka) [inline alembic/versions/a31c98c74ea0_add_e_p_v.py]: Yes, the https://api.nuget.org/packages/ itself does not represent any resource, but we use it as base uri for fetching nuget artefacts, like:

https://api.nuget.org/packages/nunit.3.7.1.nupkg
https://api.nuget.org/packages/newtonsoft.json.10.0.3.nupkg

**Comment by jpopelka** [comment]
Feel free to review & merge (if everything is green)

File: f8a_worker/workers/license.py [hunk 3]
```diff
@@ -50,13 +50,13 @@ def execute(self, arguments):
         self._strict_assert(arguments.get('ecosystem'))
         self._strict_assert(arguments.get('name'))
         self._strict_assert(arguments.get('version'))
+        eco = arguments.get('ecosystem')
+        pkg = arguments.get('name')
+        ver = arguments.get('version')
```
Comment (fridex) [inline f8a_worker/workers/license.py]: As we have strict asserts above can be directly:

```python
eco = arguments['ecosystem']
pkg = arguments['name']
ver = arguments['version']
```

**Comment by jpopelka** [comment]
Rebased, PTAL @fridex

**Pull request merged**